tempesta_fw/http_parser.c

```diff
@@ -59,6 +63,31 @@ __field_finish(TfwHttpMsg *hm, TfwStr *field,
 }
 
 /**
+ * Check whether a character is CR or LF.
+ */
+#define IS_CR_OR_LF(c) (c == '\r' || c == '\n')
+
+/**
+ * Scans the initial @n bytes of the memory area pointed to by @s for the first
+ * occurance of EOL character.
+ */
+static inline unsigned char *
+memchreol(const unsigned char *s, size_t n)
+{
+	while (n) {
+		if (IS_CR_OR_LF(*s))
+			return (unsigned char *)s;
+		s++, n--;
+	}
+	return NULL;
+}
```

Comment:
As for using `strcspn()` then according to [lxr](http://lxr.free-electrons.com/source/lib/string.c?v=4.1#L449) I don't really see that it has better implementation than `memchreol()`. Should I play with SSE 4.2 in that patch or can we separate that task from fixing CR/LF issues? Personally, I'll prefer not to mix different tasks in on patch set.